src/tests/util/fctest_geometry.F90

```diff
@@ -0,0 +1,170 @@
+! (C) Copyright 2013 ECMWF.
+! This software is licensed under the terms of the Apache Licence Version 2.0
+! which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+! In applying this licence, ECMWF does not waive the privileges and immunities
+! granted to it by virtue of its status as an intergovernmental organisation nor
+! does it submit to any jurisdiction.
+
+! This File contains Unit Tests for testing the
+! C++ / Fortran Interfaces to the Geometry Datastructure
+! @author Benjamin Menetrier
+
+#include "fckit/fctest.h"
+#include "atlas/atlas_f.h"
+
+! -----------------------------------------------------------------------------
+
+module fcta_Geometry_fixture
+use atlas_module
+use, intrinsic :: iso_c_binding
+implicit none
+end module fcta_Geometry_fixture
+
+! -----------------------------------------------------------------------------
+
+TESTSUITE_WITH_FIXTURE(fctest_atlas_Geometry,fcta_Geometry_fixture)
+
+! -----------------------------------------------------------------------------
+
+TESTSUITE_INIT
+  call atlas_library%initialise()
+END_TESTSUITE_INIT
+
+! -----------------------------------------------------------------------------
+
+TESTSUITE_FINALIZE
+  use fckit_main_module
+  call atlas_library%finalise()
+END_TESTSUITE_FINALIZE
+
+! -----------------------------------------------------------------------------
+
+TEST( test_geometry )
+use fckit_log_module
+use fckit_c_interop_module
+implicit none
+
+  type(atlas_Geometry) :: geometry
+  real(c_double) :: lonlat1(2), lonlat2(2), lonlat(2)
+  real(c_double) :: xyz1(3), xyz2(3), xyz(3)
+  real(c_double) :: distance
+
+  write(*,*) "test_geometry for UnitSphere starting"
+
+  ! Check constructor
+  geometry = atlas_Geometry("UnitSphere")
+  write(0,*) "geometry%c_ptr() = ", c_ptr_to_loc(geometry%CPTR_PGIBUG_A)
+
+  ! Define points
+  lonlat1 = (/-71.6_c_double, -33._c_double/)
+  lonlat2 = (/121.8_c_double, 31.4_c_double/)
+  xyz1 = (/2.647e-1_c_double, -7.958e-1_c_double, -5.446e-1_c_double/)
+  xyz2 = (/-4.498e-1_c_double, 7.254e-1_c_double, 5.210e-1_c_double/)
+
+  ! Check conversion from cartesian to spherical
+  call geometry%xyz2lonlat(xyz1(1), xyz1(2), xyz1(3), lonlat(1), lonlat(2))
+  FCTEST_CHECK_CLOSE( lonlat(1) , lonlat1(1), 1.e-2_c_double )
+  FCTEST_CHECK_CLOSE( lonlat(2) , lonlat1(2), 1.e-2_c_double )
+
```

Comment:
Please add tests here and below with additional requested functions in atlas_Geometry_module